### PR TITLE
feat(angular): allow nullable options

### LINKED ISSUE #387: 🚀 - Allow null in Angular directive
### Which package(s) are relevant/related to the feature request?

_No response_

### Description

Using Angular, I have a wrapped control where I want to be able to optionally pass Maskito options:

`@Input() maskito: MaskitoOptions | null = null;`

With the way the Angular directive is created, I'm not able to pass `null` as a value, so I need to use an `ngIf` to include the directive which is kind of ugly. It would be nice if `null` was a possibility and the masking would just bail early, something like:

```ts
async ngOnChanges(): Promise<void> {
    this.maskedElement?.destroy();

    if (!this.maskito) {
        return;
    }

    const predicate = this.maskitoElement;
    const predicateResult = await predicate(this.elementRef.nativeElement);

    if (this.maskitoElement !== predicate) {
        // Ignore the result of the predicate if the
        // maskito element has changed before the predicate was resolved.
        return;
    }

    this.ngZone.runOutsideAngular(() => {
        this.maskedElement = new Maskito(predicateResult, this.maskito);
    });
}
```

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

Closes #387

### CODE REVIEW

File: projects/angular/src/lib/maskito.cva.ts [hunk 1]
```diff
@@ -20,8 +20,16 @@ import {MASKITO_DEFAULT_OPTIONS, MaskitoOptions, maskitoTransform} from '@maskit
...
 export class MaskitoCva {
+    private options?: MaskitoOptions | null;
+
     @Input()
-    maskito: MaskitoOptions = MASKITO_DEFAULT_OPTIONS;
+    set maskito(options: MaskitoOptions | null | undefined) {
+        this.options = options;
+    }
+
+    get maskito(): MaskitoOptions {
```
Comment (nsbarsukov) [inline projects/angular/src/lib/maskito.cva.ts:30]: Do we really require getter/setter ? 🤔 

Could we just do the following ?

```ts
@Input()
maskito: MaskitoOptions | null = null;

accessor.writeValue = (value: unknown) => {
            original(maskitoTransform(String(value ?? ''), this.maskito ?? MASKITO_DEFAULT_OPTIONS));
        };
```

File: projects/angular/src/lib/maskito.directive.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export class MaskitoDirective implements OnDestroy, OnChanges {
     private maskedElement: Maskito | null = null;
 
     @Input()
-    maskito: MaskitoOptions = MASKITO_DEFAULT_OPTIONS;
+    maskito?: MaskitoOptions | null = MASKITO_DEFAULT_OPTIONS;
```
Comment (nsbarsukov) [inline projects/angular/src/lib/maskito.directive.ts:26]: Why do we need `undefined` ?

File: projects/angular/src/lib/maskito.cva.ts [hunk 3]
```diff
@@ -20,8 +20,16 @@ import {MASKITO_DEFAULT_OPTIONS, MaskitoOptions, maskitoTransform} from '@maskit
     },
 })
 export class MaskitoCva {
+    private options?: MaskitoOptions | null;
+
     @Input()
-    maskito: MaskitoOptions = MASKITO_DEFAULT_OPTIONS;
+    set maskito(options: MaskitoOptions | null | undefined) {
```
Comment (nsbarsukov) [inline projects/angular/src/lib/maskito.cva.ts:26]: Why do we need `undefined` ?

File: projects/angular/src/lib/maskito.pipe.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@ import {MaskitoOptions, maskitoTransform} from '@maskito/core';
     name: 'maskito',
 })
 export class MaskitoPipe implements PipeTransform {
-    transform(value: unknown, maskitoOptions: MaskitoOptions): string {
+    transform(value: unknown, maskitoOptions?: MaskitoOptions | null): string {
```
Comment (nsbarsukov) [inline projects/angular/src/lib/maskito.pipe.ts:8]: Why do we need `undefined` ?

File: projects/angular/src/lib/maskito.spec.ts [hunk 5]
```diff
@@ -58,6 +58,15 @@ describe('Maskito Angular package', () => {
...
         expect(getValue()).toBe('12345,67');
+
+        fixture.componentInstance.options = null;
+        fixture.detectChanges();
+
+        fixture.componentInstance.control.setValue(123456.9999);
+        fixture.detectChanges();
+
+        expect(getText()).toBe('123456.9999');
+        expect(getValue()).toBe('123456.9999');
```
Comment (nsbarsukov) [inline projects/angular/src/lib/maskito.spec.ts:69]: Should it be the separate `it('disable mask formatting if options is null')` ?

File: projects/core/src/lib/utils/transform.ts [hunk 6]
```diff
@@ -3,6 +3,10 @@ import {MASKITO_DEFAULT_OPTIONS} from '../constants';
 import {ElementState, MaskitoOptions} from '../types';
 import {maskitoPipe} from './pipe';
 
+export function maskitoTransform(
```
Comment (nsbarsukov) [inline projects/core/src/lib/utils/transform.ts:6]: I am not sure that we need to change signature of `maskitoTransform` 🤔

File: projects/angular/src/lib/maskito.cva.ts (hunk 1 above)
Comment (nsbarsukov) [reply projects/angular/src/lib/maskito.cva.ts:30]: Opps, my mistake, sorry.

We should keep the link to the same object even if input prop is changed.

**Review by waterplea — APPROVED** [review]

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**